app/src/main/java/monster/minions/binocularss/activities/MainActivity.kt

```diff
@@ -179,46 +241,216 @@ class MainActivity : ComponentActivity() {
+                        builder = {
+                            // Placeholder when the image hasn't loaded yet.
+                            placeholder(R.drawable.ic_launcher_foreground)
+                        }
+                    ),
+                    contentScale = ContentScale.Crop,
+                    contentDescription = description,
+                    colorFilter = if (imageExists) null else ColorFilter.colorMatrix(
+                        grayScaleMatrix
+                    ),
+                    modifier = Modifier
+                        .fillMaxSize()
+                )
+            }
+        }
+    }
 
-        theme = sharedPref
-            .getString(SettingsActivity.PreferenceKeys.THEME, "System Default")
-            .toString()
-        if (!isFirstRun) {
-            themeState.value = theme
+    /**
+     * Displays a single feed in a card view format
+     * Includes a long hold function to delete a feed
+     *
+     * @param context The application context
+     * @param feed The feed to be displayed
+     */
+    @ExperimentalCoilApi
+    @Composable
+    fun FeedCard(context: Context, feed: Feed) {
+        var showDropdown by remember { mutableStateOf(false) }
+        // Location where user long pressed.
+        var offset by remember { mutableStateOf(Offset(0f, 0f)) }
+        Card(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(8.dp)
+                .pointerInput(Unit) {
+                    detectTapGestures(
+                        onLongPress = {
+                            showDropdown = true
+                            offset = it
+                        },
+                        onTap = {
+                            // TODO temporary until articleFromFeed
+                            val intent = Intent(Intent.ACTION_VIEW)
+                            intent.data = Uri.parse(feed.link)
+                            ContextCompat.startActivity(context, intent, null)
+                        }
+                    )
+                }, elevation = 4.dp
+        ) {
+            Column {
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(16.dp),
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    // Column for feed title.
+                    Column(
+                        modifier = Modifier.width(200.dp)
+                    ) {
+                        feed.title?.let { title ->
+                            Text(text = title, fontWeight = FontWeight.SemiBold)
+                        }
+                        val items = listOf("Delete")
+                        // Convert pixel to dp
+                        val xDp = with(LocalDensity.current) { (offset.x - 15).toDp() }
+                        val yDp = with(LocalDensity.current) { (offset.y - 50).toDp() }
```

Comment:
I just realized, this manual offset (of -15 for x and -50 for y) should probably be done in dp as the dp values will not change from phone to phone, whereas the pixel values will.

Reply:
I'll go ahead and fix that really quickly